apps/dataset-browser/src/app/[locale]/store-initializer.tsx

```diff
@@ -0,0 +1,36 @@
+'use client';
+
+import {useRef} from 'react';
+import {useListStore, SortBy} from 'list-store';
+
+interface Props {
+  totalCount: number;
+  offset: number;
+  limit: number;
+  query: string;
+  sortBy: SortBy;
+  selectedFilters: {[filterKey: string]: string[] | undefined} | undefined;
+}
+
+export function StoreInitializer({
+  totalCount,
+  offset,
+  limit,
+  query,
+  sortBy,
+  selectedFilters,
+}: Props) {
+  const initialized = useRef(false);
+  if (!initialized.current) {
+    useListStore.setState({
+      totalCount,
```

Comment:
Using this code for the researcher, I realized that `totalCount` won't update after initialization. The client also updates the other properties. But `totalCount` only comes from the server.